### ISSUE TITLE
recurrence through the state dictionary

### ISSUE THREAD

**Issue by david-wb** [snowkylin/ntm#23]
https://github.com/snowkylin/ntm/blob/7db406826d6109f44c61a857ef2d1aadbbec7f54/ntm/ntm_cell.py#L106-L116

It looks there is a recurrence happening on all elements in the state dictionary. Wouldn't you need
to do a `tf.stop_gradient()` on each of these items (except controller ouput) to be consistent with the paper? At the very least there should be no recurrence on the memory `M`. Do you agree?

**Comment by snowkylin** [comment]
I do not find specific instruction in the paper that some part of the gradients should be stopped. Maybe you can try your method and see how the result will be?

**Comment by david-wb** [comment]
I tried it and the model fails to learn in a reasonable time. This makes me think original NTM was fundamentally an RNN, even the one with the feed forward controller which must have used recurrence on the memory, weights, and reads. Anyway, thanks for responding.
